javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,86 @@
+let persons = [];
+let currentPerson;
+
+function getReply(command) {
+  command = command.toLowerCase();
+  const commandSplit = command.split(" ");
+  const pattern = /what is (\d+) ([+\-*/]) (\d+)/;
+
+  if (command.startsWith("hello my name is")) {
+    const name = commandSplit[commandSplit.length - 1];
+    if (persons.length === 0 || !persons.find((p) => p.name === name)) {
+      currentPerson = { name: name, todo: [] };
+      persons.push(currentPerson);
+      return `nice to meet you ${currentPerson.name}`;
+    } else {
+      currentPerson = persons[persons.findIndex((p) => p.name === name)];
+      return `nice to see you again ${currentPerson.name}!`;
+    }
+  } else if (command === "what is my name?") {
+    if (currentPerson.name) {
+      return `your name is ${currentPerson.name}`;
+    } else {
+      return "I don't know your name yet.";
+    }
+  } else if (command.startsWith("add") && command.endsWith("to my todo")) {
+    let todo = commandSplit.slice(1, -3).join(" ");
+    currentPerson.todo.push(todo);
```

Comment:
As it contains a list of todos
```suggestion
    currentPerson.todoList.push(todo);
```